protocol/messenger.go

```diff
@@ -883,33 +833,36 @@ func (m *Messenger) Start() (*MessengerResponse, error) {
 	}
 	response := &MessengerResponse{}
 
-	mailservers, err := m.allMailservers()
+	storenodes, err := m.AllMailservers()
 	if err != nil {
 		return nil, err
 	}
 
-	response.Mailservers = mailservers
-	err = m.StartMailserverCycle(mailservers)
+	err = m.setupStorenodes(storenodes)
 	if err != nil {
 		return nil, err
 	}
 
+	response.Mailservers = storenodes
+
+	m.transport.SetStorenodeConfigProvider(m)
+
 	if err := m.communityStorenodes.ReloadFromDB(); err != nil {
 		return nil, err
 	}
 
 	go m.checkForMissingMessagesLoop()
+	go m.checkForStorenodeCycleSignals()
 
 	controlledCommunities, err := m.communitiesManager.Controlled()
 	if err != nil {
 		return nil, err
 	}
 
 	if m.archiveManager.IsReady() {
-		available := m.mailserverCycle.availabilitySubscriptions.Subscribe()
 		go func() {
 			defer gocommon.LogOnPanic()
-			<-available
+			<-m.transport.OnStorenodeAvailableOneShot()
```

Comment:
We probably need some `<-ctx.Done()` here as well?
Not related to your change, but would be nice to fix on the way.

Reply:
Actually, why not `WaitForAvailableStoreNode` here? 
A function like `OnStorenodeAvailableOneShot` is quite specific. And only used here, but could easily be replaced with `WaitForAvailableStoreNode` and infinite timeout 🤔